### ISSUE TITLE
NZB.su not getting the correct search type / ID

### ISSUE THREAD

**Issue by kiakri** [theotherp/nzbhydra#635]
* What do you expect to happen? What does happen? 
When doing check Capabilities only the TV type will be pulled down. What should happen TV & Movies should be pulled down with the correct Search ID

* Have you searched for this problem? You don't have to answer, just make sure.
Yes - Only issue i've seen has been raised by me on reddit 

* Can the problem be reproduced? How?
Hydra > Config > Indexer > Add details for NZB.su

The following will either happen.

If you add the search ID / type manually you will get an XML parser error when running check Capabilities
If you run the Capabilities check without any search ID / Type only settings relating to TV will be pulled down

[nzbhydra-debuginfo-2017-07-17-22-18.zip](https://github.com/theotherp/nzbhydra/files/1156878/nzbhydra-debuginfo-2017-07-17-22-18.zip)

**Comment by theotherp** [comment]
The problem is here:
`2017-07-17 22:18:03,776 - ERROR - newznab - Thread-883 - Error getting or parsing XML: 429 Client Error: Too Many Requests for url: https://api.nzb.su/api?extended=1&amp;t=movie&amp;offset=0&amp;apikey=&lt;APIKEY&gt;&amp;limit=50&amp;imdbid=0848228`

I had the same problem yesterday when testing. I suggest you contact the nzb.su admins. I don't see this as an error on my part. The code that works fine with a dozen other indexers causes problems with theirs.

**Comment by kiakri** [comment]
I did and this was their response. 

We support anything but headphones, but if you're seeing that error its hammering the server... you get 5 connections in 6 seconds, if it goes above that you'll get temp blocked.

**Comment by kiakri** [comment]
Please see the logs below.

https://pastebin.com/stp0faFC

Specifcally this part - Don't know if this comes out due to the timeout before it. 

2017-07-19 07:50:46,663 - ERROR - newznab - Thread-39 - An error occurred while trying to test the caps of host https://api.nzb.su: IndexerAccessException: Error getting or parsing XML

2017-07-19 07:50:46,665 - ERROR - newznab - Thread-39 - Error getting or parsing caps XML. Error message: Error getting or parsing XML

Going to fire off an email to SU.

**Comment by theotherp** [comment]
Yeah well, then this won't work. I already reduced the concurrent searches to 2 but it will still be 6 requests in a couple of seconds.

I could add some more limits but that will take some work. Until then you'll have to live without nzb.su.

**Comment by kiakri** [comment]
Looks like he has built an api hammering machine. They need to limit the number of requests per second. 

I remove some of the message as i didn't want to put it publicly

**Comment by theotherp** [comment]
I've bought VIP access so I can do proper tests. Hope that is appreciated...

**Comment by kiakri** [comment]
It's always appreciated. :)

**Comment by weirdcrap** [comment]
I also appreciate it. Su is not my main indexer but it is part of my group of indexers I use.

**Comment by theotherp** [comment]
Well, their movie parsing seems to be off.

To test if IMDB ID based searches are possible I do a search for a reasonably well known and often reuploaded movie. When the percentage of wrong results is too high I don't consider the ID supported. In this case 23% of the results are wrong. That's why movie search is considered to be not supported.

I could change the threshold but don't want to do that just for this one indexer. So I guess you need to talk to the admins and get them to improve their movie parsing.

Hint: The IMDB id I use for this test is 0848228.

**Comment by kiakri** [comment]
Reply from SU

Well, that has nothing to do with the problem that you raised... The problem you raised was for not connecting because it's hammering the API with requests.  On the best day, you're going to have some incorrect results due to parsing being wrong + reused nfo files with wrong links.  But let me know if he fixes the issue you had.

**Comment by theotherp** [comment]
I already fixed the other problem by adding pauses between requests.

The threshold I use is 10%. For comparison the percentages of wrong results from indexers using IMDB IDs:
DogNZB: 4%
DrunkenSlug: 0%
FastNZB: 0%
**nzb.su:23%**
nzbs.org: 8%
omgwtfnzbs: 0%

**Comment by kiakri** [comment]
So just to clarify should i be adding IMBD & Movies as a search type and then don't run check compatibility? 

Checking compatibility only brings up TV again.

**Comment by theotherp** [comment]
Yes. But you should expect to get a lot of wrong results.

On 21 Jul 2017 00:06, "kiakri" <notifications@github.com> wrote:

So just to clarify should i be adding IMBD & Movies as a search type and
then don't run check compatibility?

Checking compatibility only brings up TV again.

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/theotherp/nzbhydra/issues/635#issuecomment-316843671>,
or mute the thread
